llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -5704,7 +5704,8 @@ bool SelectionDAG::isKnownNeverNaN(SDValue Op, bool SNaN, unsigned Depth) const
   }
   case ISD::FMINIMUM:
   case ISD::FMAXIMUM: {
-    // TODO: Does this quiet or return the origina NaN as-is?
+    if (SNaN)
+      return true;
```

Comment:
I don't think this part is covered under the existing tests, and should be done separately from the other bit. The other piece could also just be done in one step to fully replace the IEEE versions

Reply:
Yes. You are right. We should do this when we fix the behavior of these Intrinsics.